### PR TITLE
Show direct links on stream page when available (FIRST!)

### PR DESCRIPTION
**Pull request by finitud**
Currently the links on the stream page point to the original annotated document. This pull request makes these links to be direct links to the annotation.

https://trello.com/c/iipUgujB/373-direct-link-to-annotation-from-stream-card-document-link

[reopening from now closed https://github.com/hypothesis/h/pull/3550]

### CODE REVIEW

**Comment by lenazun** [comment]
you forgot to say FIRST! 👆

**Comment by finitud** [comment]
@lenazun fixed that.

File: h/static/scripts/filter/test/document-title-test.js [hunk 1]
```diff
@@ -1,75 +0,0 @@
-'use strict';
-
-var documentTitle = require('../document-title');
-
-describe('documentTitle', function() {
-
-  it('returns the title linked if the document has title and uri', function() {
```
Comment (robertknight) [inline h/static/scripts/filter/test/document-title-test.js]: I think all the tests in here except the two "escapes HTML" tests will need equivalents for the new function - hopefully they should be somewhat simpler.

**Comment by robertknight** [comment]
After the refactoring I believe that `domain-domain.js` will also be unused. As with `document-title.js`, some of its associated tests will be relevant and should be adapted.

**Comment by robertknight** [comment]
LGTM

File: h/static/scripts/filter/test/document-title-test.js (hunk 1 above)
Comment (robertknight) [inline h/static/scripts/filter/test/document-title-test.js]: The revised tests look clear to me. If we had a lot more cases then using a data-driven test is a useful way to make things more concise.

**Pull request merged**